flink-connector-mysql-cdc/src/main/java/com/ververica/cdc/connectors/mysql/source/assigners/MySqlSnapshotSplitAssigner.java

```diff
@@ -152,7 +153,15 @@ private MySqlSnapshotSplitAssigner(
         this.currentParallelism = currentParallelism;
         this.alreadyProcessedTables = alreadyProcessedTables;
         this.remainingSplits = new CopyOnWriteArrayList<>(remainingSplits);
-        this.assignedSplits = assignedSplits;
+        this.assignedSplits =
+                assignedSplits.entrySet().stream()
+                        .sorted(Comparator.comparing(Map.Entry::getKey))
+                        .collect(
+                                Collectors.toMap(
+                                        Map.Entry::getKey,
+                                        Map.Entry::getValue,
+                                        (o, o2) -> o,
+                                        LinkedHashMap::new));
```

Comment:
I do not think we should sort `assignedSplits` here. `assignedSplits` will be changed at other methods and we cannot keep the order.
I think the problem is that the finished infos in `BinlogSplit` are not sorted. Maybe it should be sorted in `getFinishedSnapshotSplitInfos`.

Reply:
I review again and understand the changes. Please ignore this comment.

Reply:
When job restore from savepoint, if sort in getFinishedSnapshotSplitInfos by splitId and newly added tables are before exist tables that will casue newly added table(eg: eixst table list:[t1, t2], newly add table list:[a1, a2]) can't receive binlog data because of MySqlSourceEnumerator#sendBinlogMeta get wrong split partition by requestMetaGroupId

Reply:
@qg-lin, @ruanhang1993 , Hi, I understand why not sorted  in `getFinishedSnapshotSplitInfos` by splitId. But whether  need to sort `assignedSplits`? If not sort, any problem will occur?